merge request

ZBHdd opened the pull request:
the first commit is all code,and the second only append two Enter("\n")

Commit:
第一次提交，article和author为实体类，main为入口，file为结果

Commit:
第一次提交，article和author为实体类，main为入口，file为结果

lovemyliwu commented on homework/jiahang/article.py:
```diff
@@ -0,0 +1,13 @@
+class Article(object):
+    def __init__(self,dict):
```
dict是内置keyword, 请不要使用

lovemyliwu commented on homework/jiahang/author.py:
```diff
@@ -0,0 +1,10 @@
+class Author(object):
+    def __init__(self,dict):
```
同理

lovemyliwu commented on homework/jiahang/file.txt:
```diff
@@ -0,0 +1,100 @@
+标题:sunt aut facere repellat provident occaecati excepturi optio reprehenderit,作者Leanne Graham
```
作者后面加:

lovemyliwu commented on homework/jiahang/main.py:
```diff
@@ -0,0 +1,39 @@
+from article import Article
+from author import Author
+
+def str_to_object_list(str,object):
```
object是内置keyword, 请不要使用

lovemyliwu replied:
str同理

lovemyliwu commented on homework/jiahang/main.py:
```diff
@@ -0,0 +1,39 @@
+def article_match_author(filename):
+    clear_file = open(filename, "w")
+    clear_file.close()
+    file = open(filename, "a")
```
使用with方式, 就不用自己写close

lovemyliwu commented on homework/jiahang/main.py:
```diff
@@ -0,0 +1,39 @@
+    return lists
+
+def article_match_author(filename):
+    clear_file = open(filename, "w")
```
可以合并为一次文件打开操作

lovemyliwu commented on homework/jiahang/main.py:
```diff
@@ -0,0 +1,39 @@
+    try:
+        for article in articles:
+            for author in authors:
+                if article.is_written_by(author):
```
未处理文章找不到作者情况

lovemyliwu commented on homework/jiahang/main.py:
```diff
@@ -0,0 +1,39 @@
+            for author in authors:
+                if article.is_written_by(author):
+                    file.writelines("标题:{},作者{}\n".format(article.title, author.name))
+    except Exception as e:
```
一般不catch Exception

Commit:
1.rename dict ,str ,object to dictionary,string,class_object
2.unite operation deleting file and operation appending line
3.use keyword with
4.add else
5.delete try catch

lovemyliwu commented on homework/jiahang/main.py:
```diff
@@ -0,0 +1,34 @@
+                if article.is_written_by(author):
+                    file.writelines("标题:{},作者{}\n".format(article.title, author.name))
+                else:
+                    continue
```
continue也没处理啊, 建议把作者名写作:`佚名`

lovemyliwu reviewed (changes requested):
check comments

Commit:
1.after "作者" adding ":"
2.if there is no author,add "佚名"

lovemyliwu commented on homework/jiahang/file.txt:
```diff
@@ -1,100 +1,1000 @@
+标题:at nam consequatur ea labore ea harum,作者:佚名
+标题:at nam consequatur ea labore ea harum,作者:佚名
+标题:at nam consequatur ea labore ea harum,作者:佚名
+标题:at nam consequatur ea labore ea harum,作者:Clementina DuBuque
```
内容不对, 你将作者和文章进行了乘积, 而不是匹配

lovemyliwu reviewed (changes requested):
check comment

Commit:
add "break"

lovemyliwu commented on homework/jiahang/main.py:
```diff
@@ -0,0 +1,33 @@
+            for author in authors:
+                author_name = author.name if article.is_written_by(author) else "佚名"
+                file.writelines("标题:{},作者:{}\n".format(article.title, author_name))
+                break
```
任意一个文章只尝试匹配了第一作者, 其他作者呢?

lovemyliwu reviewed (changes requested):
check comment

Commit:
更改逻辑
